src/uct/sm/self/self.c

```diff
@@ -463,7 +463,7 @@ static uct_component_t uct_self_component = {
     },
     .cm_config          = UCS_CONFIG_EMPTY_GLOBAL_LIST_ENTRY,
     .tl_list            = UCT_COMPONENT_TL_LIST_INITIALIZER(&uct_self_component),
-    .flags              = UCT_MD_FLAG_RKEY_PTR,
+    .flags              = UCT_MD_FLAG_REG | UCT_MD_FLAG_RKEY_PTR,
```

Comment:
but we have this:
https://github.com/openucx/ucx/blob/1ffd1b5582cb3ccded0a2fd6072629fa7dc67bc1/src/uct/sm/self/self.c#L395

Reply:
> For some reason ucx didn't recognized self as rndv/rkey_ptr implementation without it, will check again to make sure

Maybe UCP protov2 is checking the wrong flags.
BTW, can you try removing UCT_MD_FLAG_NEED_RKEY flag for self md?